charts.py

```diff
@@ -0,0 +1,230 @@
+import logging
+import pandas
+import plotly
+import numpy
+import src
+import app
+from app import ZKLEND, NOSTRA_ALPHA, NOSTRA_MAINNET
+
+
+def get_charts_data(
+    main_chart_data, loans_data, protocols, collateral_token, debt_token
+):
+    for protocol in protocols:
+        protocol_main_chart_data = get_protocol_main_data()[protocol]
+        if protocol_main_chart_data is None or protocol_main_chart_data.empty:
+            logging.warning(
+                f"No data for pair {debt_token} - {collateral_token} from {protocol}"
+            )
+            continue
+
+        protocol_loans_data = get_protocol_loans_data()[protocol]
+
+        if main_chart_data.empty:
+            main_chart_data = protocol_main_chart_data
+            main_chart_data[f"liquidable_debt_{protocol}"] = protocol_main_chart_data[
+                "liquidable_debt"
+            ]
+            main_chart_data[f"liquidable_debt_at_interval_{protocol}"] = (
+                protocol_main_chart_data["liquidable_debt_at_interval"]
+            )
+        else:
+            main_chart_data["liquidable_debt"] += protocol_main_chart_data[
+                "liquidable_debt"
+            ]
+            main_chart_data["liquidable_debt_at_interval"] += protocol_main_chart_data[
+                "liquidable_debt_at_interval"
+            ]
+            main_chart_data[f"liquidable_debt_{protocol}"] = protocol_main_chart_data[
+                "liquidable_debt"
+            ]
+            main_chart_data[f"liquidable_debt_at_interval_{protocol}"] = (
+                protocol_main_chart_data["liquidable_debt_at_interval"]
+            )
+
+        if loans_data.empty:
+            loans_data = protocol_loans_data
+        else:
+            loans_data = pandas.concat([loans_data, protocol_loans_data])
+
+        return main_chart_data, loans_data
+
+
+def get_protocol_main_data(
```

Comment:
name of the function should return the same in return
You have `protocol_main_chart_data_mapping` as return but function name is `get_protocol_main_data`
I would rename function name to : `get_protocol_mapping_chart_data` or something like this